client/my-sites/stats/site.jsx

```diff
@@ -173,14 +174,13 @@ module.exports = React.createClass( {
 								period={ this.props.period }
 								date={ queryDate }
 								beforeNavigate={ this.updateScrollPosition } />
-							<StatsModule
-								path={ 'referrers' }
+							<StatsConnectedModule
+								path="referrers"
 								moduleStrings={ moduleStrings.referrers }
-								site={ site }
-								dataList={ this.props.referrersList }
 								period={ this.props.period }
+								query={ query }
 								date={ queryDate }
-								beforeNavigate={ this.updateScrollPosition } />
+								statType="statsReferrers" />
```

Comment:
We need to add `showSummaryLink={ true }` prop here so the "View All" link in the footer will be shown when applicable.